spec/unleash/constraint_spec.rb

```diff
@@ -72,5 +72,306 @@
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_STARTS_WITH', ['ment'])
+      expect(constraint.matches_context?(context)).to be_falsey
+    end
+
+    it 'matches based on property STR_ENDS_WITH value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: 'development'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'STR_ENDS_WITH', ['ment'])
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_ENDS_WITH', ['development'])
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_ENDS_WITH', ['dev'])
+      expect(constraint.matches_context?(context)).to be_falsey
+    end
+
+    it 'matches based on property STR_CONTAINS value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: 'development'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['ment'])
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['dev'])
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['development'])
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'STR_CONTAINS', ['DEVELOPMENT'])
+      expect(constraint.matches_context?(context)).to be_falsey
+    end
+
+    it 'matches based on property NUM_EQ value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: '3.141'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'NUM_EQ', '3.141')
+      expect(constraint.matches_context?(context)).to be_truthy
+
+      constraint = Unleash::Constraint.new('env', 'NUM_EQ', '2.718')
+      expect(constraint.matches_context?(context)).to be_falsey
+    end
+
+    it 'matches based on property NUM_LT value' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: '3.141'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+
+      constraint = Unleash::Constraint.new('env', 'NUM_LT', '2.718')
+      expect(constraint.matches_context?(context)).to be_falsey
```

Comment:
I think it is fair that we expect these new tests to return strict `true` or `false` values, no?
```suggestion
      expect(constraint.matches_context?(context)).to be false
```

Reply:
Good call, done!